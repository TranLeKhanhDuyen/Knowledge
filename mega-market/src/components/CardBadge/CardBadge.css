.card-badge-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: var(--primary-color);
  width: fit-content;
  padding: 11px 8px;
  border-radius: 0px 16px 0px 16px;
}

.card-badge {
  font-size: var(--font-md);
  font-weight: var(--fw-bold);
  color: var(--white-color);
  text-transform: uppercase;
}
