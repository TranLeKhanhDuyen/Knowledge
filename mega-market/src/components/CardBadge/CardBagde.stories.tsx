import type { Meta, StoryObj } from '@storybook/react'

import { CardBadge } from '@components'

export default {
  title: 'Components/CardBadge',
  component: CardBadge,
  tags: ['autodocs'],
  argTypes: {
    additionalClass: { description: 'Add class to the button' },
    title: { description: 'title content of button.' },
    percent: { description: 'percent content of button.' }
  }
} as Meta

type Story = StoryObj<typeof CardBadge>

export const Default: Story = {
  args: {
    percent: '50%',
    title: 'off'
  }
}
