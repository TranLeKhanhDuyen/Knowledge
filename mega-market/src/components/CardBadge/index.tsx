import React from 'react'
import './CardBadge.css'

interface ICardBadgeProps extends React.HTMLAttributes<HTMLDivElement> {
  percent: string
  title: string
  additionalClass?: string
}

const CardBadge: React.FC<ICardBadgeProps> = ({
  percent,
  title,
  additionalClass = '',
  ...restProps
}) => {
  return (
    <aside className={`card-badge-container ${additionalClass}`} {...restProps}>
      <p className={`card-badge card-badge-percent`}>{percent}</p>
      <p className={`card-badge card-badge-title`}>{title}</p>
    </aside>
  )
}

export default CardBadge
