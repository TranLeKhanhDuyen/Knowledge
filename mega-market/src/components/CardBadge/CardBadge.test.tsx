import { render } from '@testing-library/react'
import CardBadge from '.'

describe('CardBadge component', () => {
  test('renders correctly', () => {
    const { container } = render(<CardBadge percent='50' title='Test Title' />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders percent and title correctly', () => {
    const percent = '50'
    const title = 'Test Title'
    const { getByText } = render(<CardBadge percent={percent} title={title} />)
    expect(getByText(`${percent}%`)).toBeInTheDocument()
    expect(getByText(title)).toBeInTheDocument()
  })

  test('applies additionalClass correctly', () => {
    const additionalClass = 'additional-class'
    const { container } = render(
      <CardBadge
        percent='50'
        title='Test Title'
        additionalClass={additionalClass}
      />
    )
    expect(container.firstChild).toHaveClass(
      'card-badge-container',
      additionalClass
    )
  })
})
