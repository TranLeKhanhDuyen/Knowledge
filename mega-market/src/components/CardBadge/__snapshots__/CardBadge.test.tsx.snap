// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBadge component renders correctly 1`] = `
<aside
  class="card-badge-container "
>
  <p
    class="card-badge badge-percent"
  >
    50
    %
  </p>
  <p
    class="card-badge badge-title"
  >
    Test Title
  </p>
</aside>
`;
